ibc-core/ics03-connection/types/src/connection.rs

```diff
@@ -465,7 +460,8 @@ impl Counterparty {
 )]
 #[cfg_attr(
     feature = "borsh",
-    derive(borsh::BorshSerialize, borsh::BorshDeserialize)
+    derive(borsh::BorshSerialize, borsh::BorshDeserialize),
+    borsh(use_discriminant = false)
```

Comment:
can you comment some details on this? just for my context.

Reply:
[This](https://docs.rs/borsh/latest/borsh/derive.BorshSerialize.html#2-borshuse_discriminantbool-item-level-attribute) is the relevant section in the `borsh` docs. Any enums with explicit discriminants need to specify either `use_discriminant = false`, which utilizes `borsh` v0.10.3's discriminant serialization scheme, or `use_discriminant = true` to serialize the discriminants as-is. 

I went with `use_discriminant = false` since we were just on `borsh` v0.10 and figured that was the safer option to go for.

Reply:
hm interesting. Is this being using in `ibc-proto` too? If not, can we circle back with composable team and maybe remove this?

Reply:
> Is this being using in ibc-proto

I don't recall getting any compiler warnings telling me this attribute needed to be added when I updated `borsh` in ibc-proto.

> can we circle back with composable team

I'm not entirely clear on why this is relevant to them

> and maybe remove this

I'm not sure it can be. When I updated the dependencies, the compiler surfaced an error saying that this attribute needed to be added.

Reply:
Ok. let's go with this then :)